website/docs/usage/saving-loading.md

```diff
@@ -203,11 +203,15 @@ the data to and from a JSON file.
 
 ```python
 ### {highlight="16-23,25-30"}
+import json
+from spacy import Language
 from spacy.util import ensure_path
 
 @Language.factory("my_component")
 class CustomComponent:
-    def __init__(self):
+    def __init__(self, nlp: Language, name: str = "my_component") -> None:
+        self.nlp = nlp
+        self.name = name
```

Comment:
Just a couple minor simplifications for the purpose of the example: you'd only need to keep a reference to `nlp` if you need access to the full pipeline within the component, which is rare, and typically at most you'd want to have the vocab (`self.vocab = nlp.vocab`).

```suggestion
    def __init__(self, nlp: Language, name: str = "my_component"):
        self.name = name
```

Reply:
I confirm that it's working with your simplification, thanks for the feedback!